lib/matplotlib/widgets.py

```diff
@@ -104,7 +104,10 @@ def _changed_canvas(self):
            True if the canvas has been changed.
 
         """
-        return self.canvas is not self.ax.figure.canvas
+        # no canvas, something is very fishy
+        if self.canvas is None:
+            return True
+        return self.canvas is not self.canvas.figure.canvas
```

Comment:
Can be made one-liner as `return self.canvas is not None and self.canvas is not self.canvas.figure.canvas`.